src/runtime_src/core/tools/xbtracer/src/app/launcher.cpp

```diff
@@ -0,0 +1,534 @@
+          "-bit but target application is ", (is_child_wow64 ? w64 : w32),
+          "-bit");
+    return false;
+  }
+
+  return true;
+}
+
+// Function to set an environment variable
+bool SetEnvVariable(const std::string& name, const std::string& value)
+{
+  LPCSTR lpName = name.c_str();
+  LPCSTR lpValue = value.c_str();
+
+  BOOL result = SetEnvironmentVariableA(lpName, lpValue);
+
+  if (result == 0) {
+    DWORD errorCode = GetLastError();
+    std::cerr << "Failed to set environment variable. Error code: " << errorCode
+              << std::endl;
+    return false;
+  }
+
+  return true;
+}
+
+/*
+ * Function to get the current precise system time as pair for string and
+ * formated string
+ */
+std::pair<std::string, std::string> GetCurrentTimeAsString()
+{
+  FILETIME ft;
+  GetSystemTimePreciseAsFileTime(&ft);
+
+  ULARGE_INTEGER uli;
+  uli.LowPart = ft.dwLowDateTime;
+  uli.HighPart = ft.dwHighDateTime;
+
+  // Convert the time to a string
+  std::ostringstream oss;
+  oss << uli.QuadPart;
+
+  // Time formatting and trace directory printing
+  FILETIME localFt;
+  FileTimeToLocalFileTime(&ft, &localFt);
+
+  SYSTEMTIME st;
+  FileTimeToSystemTime(&localFt, &st);
+
+  std::ostringstream ss;
+  ss << std::setfill('0') << std::setw(4) << st.wYear << "-" << std::setw(2)
+     << st.wMonth << "-" << std::setw(2) << st.wDay << "_" << std::setw(2)
+     << st.wHour << "-" << std::setw(2) << st.wMinute << "-" << std::setw(2)
+     << st.wSecond;
+
+  std::string formattedTime = ss.str();
+
+  return {oss.str(), formattedTime};
+}
+#else
+/* Warning: variable 'environ' is non-const and globally accessible
+ * It ensures compatibility with existing systems and libraries that rely on
+ * environ. Its usage is typically limited and encapsulated, reducing the risk
+ * of misuse.
+ */
+// NOLINTNEXTLINE (cppcoreguidelines-avoid-non-const-global-variables)
+extern char **environ;
+#endif
+
+/*
+ * Main Function
+ */
+int main(int argc, char* argv[])
+{
+  int option = 0;
+
+  launcher& app = launcher::getInstance();
+  bool inst_debug = false;
+  std::vector<std::string> args(argv, argv + argc);
+#ifdef _WIN32
+  DWORD retval = 0;
+
+  strncpy_s(app.m_name.data(), app.m_name.size(), PathFindFileName(argv[0]), _TRUNCATE);
+#else
+  int retval = 0;
+  std::string name = "";
+  std::filesystem::path path(args[0]);
+  if (!path.filename().empty()) {
+    name = path.filename().string();
+  }
+
+  if (!name.empty()) {
+    /* Warning: do not implicitly decay an array into a pointer
+     * The array-to-pointer decay is necessary and controlled for interfacing
+     * with C-style functions.
+     * Size control ensures that the strncpy operation is safe and prevents
+     * buffer overflows.
+     */
+    // NOLINTNEXTLINE (cppcoreguidelines-pro-bounds-array-to-pointer-decay,hicpp-no-array-decay)
+    strncpy(app.m_name.data(), name.c_str(), str_sz_s);
+  }
+#endif
+  try {
+    std::lock_guard<std::mutex> lock(getopt_mutex);
```

Comment:
Looks like getopt_mutex is used exclusively here.  So do this and remove the global.
```suggestion
    static std::mutex mutex;
    std::lock_guard lock(mutex);
```

Reply:
Done